src/main/io/osd.c

```diff
@@ -258,12 +260,34 @@ static void osdDrawSingleElement(uint8_t item)
         }
 
 #ifdef USE_RTC6705
+
+#endif
+
+// This will eventually be gone.
+#ifdef USE_RTC6705
         case OSD_VTX_CHANNEL:
         {
             sprintf(buff, "CH:%d", current_vtx_channel % CHANNELS_PER_BAND + 1);
             break;
         }
-#endif // VTX
+#endif // USE_RTC6705
+
+#ifdef VTX_COMMON
+        case OSD_VTX_CHANNEL:
+        {
+            // Some devices may be operating in non-band/chan mode,
+            // in which case current operating frequency is displayed.
+            char *str;
+            if ((str = vtxCommonGetBandChanString()) != NULL) {
+                sprintf(buff, "CH:%s", vtxCommonGetBandChanString());
+            } else if ((str = vtxCommonGetBandChanString()) != NULL) {
```

Comment:
Also, isn't it more flash-efficient to just declare "CH:--" and let `vxtCommonGetBandChanString` overwrite the `--` ? I might be wrong, but I remember `sprintf` being quite slow because of the parsing.